MergerLogic/ImageProcessing/TileScaler.cs

```diff
@@ -9,25 +9,52 @@ public class TileScaler : ITileScaler
         private const int TILE_WIDTH = 256;
         private const int TILE_HEIGHT = 256;
 
-        public void Upscale(MagickImage baseImage, Tile baseTile, Coord targetCoords)
+        public MagickImage Upscale(MagickImage baseImage, Tile baseTile, Coord targetCoords)
         {
             int zoomLevelDiff = targetCoords.Z - baseTile.Z;
             int scale = 1 << zoomLevelDiff;
-            double scaleAsDouble = (double)scale;
 
-            double tilePartX = (targetCoords.X % scale) / scaleAsDouble;
-            double tilePartY = (targetCoords.Y % scale) / scaleAsDouble;
-            double tileSize = TILE_HEIGHT / scaleAsDouble;
+            //find source pixels range
+            double tilePartX = targetCoords.X % scale;
+            double tilePartY = targetCoords.Y % scale;
+            double tileSize = TILE_HEIGHT / (double)scale;
 
-            int pixleX = (int)(tilePartX * TILE_WIDTH);
-            int pixleY = (int)(tilePartY * TILE_HEIGHT);
-            int imageWidth = Math.Max((int)tileSize, 1);
-            int imageHeight = Math.Max((int)tileSize, 1);
+            int pixelX = (int)(tilePartX * tileSize);
+            int pixelY = (int)(tilePartY * tileSize);
+            int srcSize = Math.Max((int)tileSize, 1);
+            int maxSrcX = pixelX + srcSize;
+            int maxSrcY = pixelY + srcSize;
 
-            MagickGeometry geometry = new MagickGeometry(pixleX, pixleY, imageWidth, imageHeight);
-            baseImage.Crop(geometry);
-            baseImage.RePage();
-            baseImage.Resize(TILE_WIDTH, TILE_HEIGHT);
+            //prepare pixels data 
+            var scaledImage = new MagickImage(MagickColor.FromRgba(0, 0, 0, 0), TILE_WIDTH, TILE_HEIGHT);
+            scaledImage.HasAlpha = baseImage.HasAlpha;
+            var srcPixels = baseImage.GetPixels();
+            var targetPixels = scaledImage.GetPixels();
+            int byteCount = 4 * scale * scale;
```

Comment:
4 times the square of the zoom level scale difference? why?

Reply:
4 color channels (RGBA), with 1 byte each, and both x and y axis are multiplied by the scale so it is squared